pkg/assets/builder.go

```diff
@@ -75,13 +79,28 @@ func (a *AssetBuilder) remapImage(image string) (string, error) {
 		// 3. make kops and create/apply cluster
 		override := os.Getenv("DNSCONTROLLER_IMAGE")
 		if override != "" {
-			image = override
+			asset.Mirror = override
+			a.Assets = append(a.Assets, asset)
+			return image, nil
 		}
 	}
 
-	asset.Mirror = image
+	if strings.HasPrefix(image, GCR_IO) {
+		override, err := GetGoogleImageRegistryContainer(a.ClusterSpec, image)
```

Comment:
The reason is legacy code.  We have methods that were just calling a method to get a google container repository. https://github.com/kubernetes/kops/blob/master/pkg/model/components/context.go#L122 <- this method specifically.  So I created a method to deal with google containers specifically.  We can file an issue to refactor because this code is now spread across at least 3 PRs.

Frankly, I wrote this code about a month ago, so that may not be the reason.  If it is a change here, please just ask me to make the change.